File: java-web-spring-boot-3/src/main/java/com/vndevteam/javawebspringboot3/infrastructure/util/MsgUtils.java
package com.vndevteam.javawebspringboot3.infrastructure.util;

import com.vndevteam.javawebspringboot3.domain.enums.Message;
import org.springframework.context.MessageSource;
import org.springframework.context.i18n.LocaleContextHolder;

public class MsgUtils {
    private static MessageSource messageSource;

    public static void setMessageSource(MessageSource messageSource) {
        MsgUtils.messageSource = messageSource;
    }

    public static String getMessage(Message code) {
        return messageSource.getMessage(code.code, null, LocaleContextHolder.getLocale());
    }

    public static String getMessage(Message code, Object... args) {
        return messageSource.getMessage(code.code, args, LocaleContextHolder.getLocale());
    }
}

File: kotlin-webmvc-spring-boot-3/src/main/kotlin/com/vndevteam/kotlinwebmvcspringboot3/infrastructure/util/MsgUtils.kt
package com.vndevteam.kotlinwebmvcspringboot3.infrastructure.util

import com.vndevteam.kotlinwebmvcspringboot3.domain.enums.MESSAGE
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder

object MsgUtils {
    private lateinit var messageSource: MessageSource

    fun setMessageSource(messageSource: MessageSource) {
        this.messageSource = messageSource
    }

    fun getMessage(code: MESSAGE): String {
        return messageSource.getMessage(code.code, null, LocaleContextHolder.getLocale())
    }

    fun getMessage(code: MESSAGE, vararg args: Any): String {
        return messageSource.getMessage(code.code, args, LocaleContextHolder.getLocale())
    }
}
